native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -3617,12 +3618,18 @@ static int proxy_cluster_pre_request(proxy_worker **worker,
 
                 return HTTP_SERVICE_UNAVAILABLE;
             } else {
-                ap_log_error(APLOG_MARK, APLOG_DEBUG, 0, r->server,
+                int loglevel = APLOG_DEBUG;
+                 apr_status_t error_code = HTTP_NOT_FOUND;
+                if (return503_when_no_context) {
+                    loglevel = APLOG_ERR;
+                    error_code = HTTP_SERVICE_UNAVAILABLE;
+                }
```

Comment:
The indentation is not ok here (additional space).

I would personally write
```c
int log_level = return503_when_no_context ? APLOG_ERR : APLOG_DEBUG;
error_code = return503_when_no_context ? HTTP_SERVICE_UNAVAILABLE : HTTP_NOT_FOUND;
```

but that's a matter of preference.